EIPS/EnsIP01.md

```diff
@@ -0,0 +1,63 @@
+
+## Overview
+
+The purpose of this document is to explore a way in which the properties of a blind vickrey auction can be kept, while removing the requirement of the user to reveal it a posteriori, by making bids as time-locked messages that can be "mined" by a network of computers. The process would be as follows:
+
+1) **Encoding the bid** The potential bidder decides a name and then creates a random private key, with a small amount of bytes according to a **difficulty factor** given by the contract. Then it uses this small key to encrypt a message containing the name it wants and the amount it is willing to pay for it.
+
+2) **Bidding** the bidder then makes a transaction to the chain with an amount of ether equal to or larger than their secret price. The contract doesn't need to store the full message, only a hash of it, the rest of the message can either be given off-chain to "miners" or it can be logged into an event on-chain to ensure it's available for everyone. The bidder also must reveal the size of the key they picked.
+
+3) **Unlocking** anyone is able to "reveal" the bid by revealing the private key. It can be done by the bidder themselves or by other machines by bruteforcing them. Since the owner has revealed the size of the key, machines can decide not to try to unlock a message if they believe it will not be able to do so
+
+4) **difficulty adjustments** If a key is found in less than 48 hours, the **difficulty** factor is slightly increased, in order to target a median of 48 hours. And in the same manner, whenever a bid is revealed after 48h the difficulty should also decrease ever so slightly
+
+5) **Rewards** Rewards are given to the first person to reveal the key, so anyone bruteforcing it should reveal theirs as soon as they find one, or they might lose the chance of a reward. If no one has found any key after 48h, then the reward should increase exponentially, until a maximum period of 120 hours. After that period, it can only be revealed by the owner, and if the auction has been won by someone else, the bid will be declared invalid. If a large amount of bids go unrevealed, then the rewards should also adjust automatically.
+
+6) **Minimum deposits** The current system has a minimum deposit hard coded to 0.01. This was never considered a final solution because of the floating value of ether and demand for ENS names. In this method the minimum deposit would be proportional to the revealer's fees. A value of twice the reveal fee would be suggested.
+
+7) **Invalid bids** one extra factor would go into checking if a bid was valid: if the bid was revealed and the private key was larger than the size they had revealed, then the bid is declared invalid. Users are free to pick keys larger than the difficulty if they want to reveal it themselves, but they must be honest on the minimum size, so that revealers can choose to ignore bids that would take longer than 24 hours.
```

Comment:
What if they're not honest? There's no mechanism here to either prove that a key is bruteforceable, or to punish those who make the miners do useless work.

Reply:
I think a minimum requirement should be a way to prove the length of the private key. This provides an upper bound for difficulty. This might be as simple as providing the public key and signing a message (to prove that there actually is a key).

Reply:
@Arachnid  This is why having some sort of minimum deposit is important, to create a cost of unrevealed invalid bids. 

@Sjors I agree completely, but I don't know if there is any way to prove the length of a private key